app-backend/db/src/test/scala/com/azavea/rf/database/ProjectDaoSpec.scala

```diff
@@ -274,4 +274,197 @@ class ProjectDaoSpec extends FunSuite with Matchers with Checkers with DBTestCon
+      forAll {
+        ( userTeamOrgPlat: (User.Create, Team.Create, Organization.Create, Platform),
+          acr1: ObjectAccessControlRule,
+          acr2: ObjectAccessControlRule,
+          project: Project.Create,
+          grantedUser: User.Create
+        ) => {
+          val projectDeletePermissionsIO = for {
+            projMiscInsert <- fixUpProjMiscInsert(userTeamOrgPlat, project)
+            (projectInsert, dbUserTeamOrgPlat) = projMiscInsert
+            dbGrantedUser <- UserDao.create(grantedUser)
+            dbUserGrantedTeamOrgPlat = dbUserTeamOrgPlat.copy(_1 = dbGrantedUser)
+            acrInsert1 = fixUpObjectAcr(acr1, dbUserGrantedTeamOrgPlat)
+            acrInsert2 = fixUpObjectAcr(acr2, dbUserGrantedTeamOrgPlat)
+            permsAdded <- ProjectDao.addPermissionsMany(projectInsert.id, List(
+              acrInsert1,
+              acrInsert2
+            ))
+            permsDeleted <- ProjectDao.deletePermissions(projectInsert.id)
+          } yield { (permsAdded, List(acrInsert1, acrInsert2).map(Some(_)), permsDeleted) }
+
+          val (permsAdded, acrsAdded, permsDeleted) = projectDeletePermissionsIO.transact(xa).unsafeRunSync
+
+          assert(permsAdded.diff(acrsAdded).length == 0,
+            "Inserting permissions to a project should get them back")
+          assert(permsDeleted.length == 0,
+            "Deleting project permissions should get an empty list back")
+          true
+        }
+      }
+    }
+  }
+
+  // listUserActions
+  test("add permissions to a project and list user actions") {
+    check {
+      forAll {
+        ( userTeamOrgPlat: (User.Create, Team.Create, Organization.Create, Platform),
+          acr1: ObjectAccessControlRule,
+          acr2: ObjectAccessControlRule,
+          acr3: ObjectAccessControlRule,
+          project: Project.Create,
+          grantedUser: User.Create
+        ) => {
+          val userActionsIO = for {
+            projMiscInsert <- fixUpProjMiscInsert(userTeamOrgPlat, project)
+            (projectInsert, dbUserTeamOrgPlat) = projMiscInsert
+            (dbUser, dbTeam, dbOrg, dbPlatform) = dbUserTeamOrgPlat
+            dbGrantedUser <- UserDao.create(grantedUser)
+            dbUserGrantedTeamOrgPlat = dbUserTeamOrgPlat.copy(_1 = dbGrantedUser)
+            _ <- {
+              UserGroupRoleDao.create(UserGroupRole.Create(dbGrantedUser.id, GroupType.Platform, dbPlatform.id, GroupRole.Member)
+                .toUserGroupRole(dbUser, MembershipStatus.Approved)) >>
+              UserGroupRoleDao.create(UserGroupRole.Create(dbGrantedUser.id, GroupType.Organization, dbOrg.id, GroupRole.Member)
+                .toUserGroupRole(dbUser, MembershipStatus.Approved)) >>
+              UserGroupRoleDao.create(UserGroupRole.Create(dbGrantedUser.id, GroupType.Team, dbTeam.id, GroupRole.Member)
+                .toUserGroupRole(dbUser, MembershipStatus.Approved))
+            }
+            acrInsert1 = fixUpObjectAcr(acr1, dbUserGrantedTeamOrgPlat)
```

Comment:
You can `map(fixupObjectAcr(_, dbUserGratnedTeamOrgPlat)` on a list of the inserts here if you want, or generate
a list of them to begin with and `map` on that -- we've done custom list generators elsewhere (see e.g. `Gen[List[Annotation.Create]]` to avoid the really long lists that scalacheck wants to create if left to its own devices.

Reply:
Though I see that would require some rework of the test condition, so feel free to ignore this one.